sympy/geometry/point.py

```diff
@@ -106,7 +106,7 @@ def __new__(cls, *args, **kwargs):
     is_Point = True
 
     def __contains__(self, item):
-        return item == self
+        return item in self.args
```

Comment:
I think

    p = Point(1,2)
    x,y = p
    l = list(p)
    p2 = Point(x**2 for x in p)

are pretty good reasons for a point to be iterable.